domain/src/main/java/org/oppia/android/domain/exploration/ExplorationProgressController.kt

```diff
@@ -698,15 +698,26 @@ class ExplorationProgressController @Inject constructor(
         val ephemeralState = computeBaseCurrentEphemeralState()
         when {
           answerOutcome.destinationCase == AnswerOutcome.DestinationCase.STATE_NAME -> {
-            endState()
-            val newState = explorationProgress.stateGraph.getState(answerOutcome.stateName)
-            explorationProgress.stateDeck.pushState(
-              newState,
-              prohibitSameStateName = true,
-              timestamp = startSessionTimeMs + continueButtonAnimationDelay,
-              isContinueButtonAnimationSeen = isContinueButtonAnimationSeen
-            )
-            hintHandler.finishState(newState)
+            val wasVisited = explorationProgress.stateDeck
+              .wasPreviouslyVisited(answerOutcome.stateName)
+
+            // Checks whether Learner submits wrong answer and destination name was previously
+            // visited by Learner.
+            if (!doesInteractionAutoContinue(answerOutcome.state.interaction.id) &&
```

Comment:
I'm actually wondering if this `doesInteractionAutoContinue` is for the current state, or the flashback state.

Reply:
This sounds good, thanks!